Storage/src/Connection/Rest.php

```diff
@@ -660,21 +653,15 @@ private function chooseValidationMethod(array $args)
     private function crcFromStream(StreamInterface $data)
     {
         $pos = $data->tell();
-
-        if ($pos > 0) {
-            $data->rewind();
-        }
-
-        $crc32c = CRC32::create(CRC32::CASTAGNOLI);
-
         $data->rewind();
+        $crc32c = hash_init('crc32c');
         while (!$data->eof()) {
-            $crc32c->update($data->read(1048576));
+            $buffer = $data->read(1048576);
+            hash_update($crc32c, $buffer);
         }
-
         $data->seek($pos);
-
-        return base64_encode($crc32c->hash(true));
+        $hash = hash_final($crc32c, true);
```

Comment:
Are we sure `hash_final` exists in all versions of PHP 7.4+? If not, we should keep the code for `google/crc32` as a fallback.

Reply:
On [PHP doc site](https://www.php.net/manual/en/function.hash-init.php), it flags (PHP 5 >= 5.1.2, PHP 7, PHP 8, PECL hash >= 1.1) and `hash` is a core extension which is always enabled since php7.4+, so unless a user builds PHP from source AND excludes hash extension (maybe to make the binary leaner) they will have these functions.

So, to be on a safe side, I added a check for the hash function and act accordingly.